lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -3018,6 +3035,21 @@ LogicalResult FIRRTLLowering::visitDecl(InstanceOp oldInstance) {
       continue;
     }
 
+    // If the result has an analog type and is used only by attach op,
+    // try eliminating a temporary wire by directly using an attached value.
+    if (portResult.getType().isa<AnalogType>()) {
+      AttachOp attach;
+      if (portResult.hasOneUse() &&
+          (attach = dyn_cast<AttachOp>(*portResult.getUsers().begin()))) {
```

Comment:
Nit: Can this be flattened out?

```suggestion
      AttachOp attach = dyn_cast<AttachOp>(*portResult.getUsers().begin());
      if (!portResult.hasOneUse() || !attach)
        continue;
```

Reply:
`if (!portResult.hasOneUse() || !attach) continue;` is incorrect because we have to lower the port anyway. I simplified the condition a bit, thanks!